contracts/drafts/classifieds/Classifieds.sol

```diff
@@ -0,0 +1,211 @@
+pragma solidity ^0.5.10;
+
+import "@openzeppelin/contracts/ownership/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC721/IERC721.sol";
+import "@openzeppelin/contracts/token/ERC721/IERC721Receiver.sol";
+import "@openzeppelin/contracts/math/SafeMath.sol";
+
+
+/**
+ * @title Classifieds
+ * @notice Implements the classifieds board market
+ */
+contract Classifieds is IERC721Receiver {
+
+    using SafeMath for uint256;
+
+    struct Ad {
+        address poster;
+        address token;
+        uint256 descriptor;
+        uint256 deadline;
+        bool isCollectible;
+        bool cancelled;
+        bool resolved;
+    }
+
+    mapping(uint256 => bytes[]) public adsByCreationDateInMin;
+
+    mapping(bytes => Ad) public adsByHash;
+
+    mapping(bytes => bytes[]) public fillersByAd;
+
+    function onERC721Received(
+        address operator,
+        address from,
+        uint256 tokenId,
+        bytes memory data
+    )
+    public returns (bytes4) {
```

Comment:
Why are you doing this?

Reply:
Oh, that's because I need to transfer from an ERC721 address to my classifieds contract, and so my contract needs to implement that method from IERC721Receiver, which just returns the specified magic number.

Reply:
did you test?
I'm still a bit confused.

Reply:
Yeah, I did some tests.

Reply:
Can't find it. Can you point me to them, please?

Reply:
Sure, it's the fillAd and resolveAd tests in test/drafts/classifieds/Classifieds.test.ts. I don't have a test function just for this function, but you can test whether you can transfer erc721 to the contract using those functions.